cmd/ddev/cmd/composer-create.go

```diff
@@ -241,3 +240,61 @@ func init() {
 	ComposerCmd.AddCommand(ComposerCreateProjectCmd)
 	ComposerCmd.AddCommand(ComposerCreateCmd)
 }
+
+// removeAll removes all files and folders in path except the ones listed in
+// skipRecursive and skip where skip only preserves the folder itself but still
+// removes containing files and folders.
+func removeAll(aPath string, skipRecursive, skip []string) error {
+	fsObjects, err := fileutil.ListFilesInDir(aPath)
+	if err != nil {
+		return err
+	}
+
+ObjectsLoop:
+	for _, fsObject := range fsObjects {
+		// Just skip objects listed in skipRecursive.
+		for _, skipObject := range skipRecursive {
+			if fsObject == skipObject {
+				continue ObjectsLoop
+			}
+		}
+
+		for _, skipObject := range skip {
+			if strings.HasPrefix(skipObject, fsObject) {
```

Comment:
These are filepaths, we should be using filepath to do this, not strings.HasPrefix(). Otherwise we can make mistakes with directories that have a partial path (prefix)